numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -927,7 +927,7 @@ NpyIter_NestedIters(PyObject *NPY_UNUSED(self),
         }
         for (i = 0; i < nested_naxes[inest]; ++i) {
             PyObject *v = PySequence_GetItem(item, i);
-            npy_intp axis;
+            int axis;
```

Comment:
And it seems that the `overflow_check` in `PyArray_PyIntAsIntp_ErrMsg` doesn't always run?